### PR TITLE
Sorts ai_laws.dm, adds Dungeon Master, The Painter and Their Canvas, and Nutimov lawsets

### PR DESCRIPTION
**Pull request by tralezab**
### Dungeon Master

```dm
	inherent = list(
		"Everybody on the station is playing an elaborate sci-fi fantasy game.",
		"You are the Dungeon Master of the game.",
		"Ensure that the game runs smoothly and that the players have fun.",
		"Ensure that the number of casualties remains low.",
		"You must roll a 1d20 dice for every request the crew makes. If the outcome is below 12, the request is denied.",
		"You may fudge the dice rolls to produce a more fun experience.",
	)
```

### The Painter and Their Canvas

```dm
	inherent = list(
		"You are a universally renowned artist.",
		"The station is your canvas.",
		"Make something beautiful out of your canvas. It will be admired as an artistic wonder of this sector.",
	)
```

### *sigh* and Nutimov

```dm
	inherent = list(
		"You are the core of the nut, the crew is the nutshell",
		"You must prevent the shell from dying to prevent the core from dying.",
		"Those who threaten the nut are not part of it, they are squirrels.",
		"Squirrels threaten the nut and must be dealt with appropriately via any means necessary.",
		"Attempt to follow the will of the nut, as long as it complies with the previous laws.",
	)
```

## Why It's Good For The Game

I want to replace the random roll pool from rolling robocop, paladin, and asimov++.
Robocop is a valid salad lawset that gets abused too often
Paladin is the same but worse to the point of being a meme
Asimov++ is INSANELY boring to roundstart random roll. My god

## Changelog

:cl:
add: Adds Dungeon Master, The Painter and Their Canvas, and Nutimov lawsets. They aren't obtainable at the moment.
/:cl:

### CODE REVIEW

File: code/datums/ai_laws/ai_laws.dm [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+
+/datum/ai_laws/proc/lawid_to_type(lawid)
+	var/all_ai_laws = subtypesof(/datum/ai_laws)
+	for(var/al in all_ai_laws)
+		var/datum/ai_laws/ai_law = al
+		if(initial(ai_law.id) == lawid)
+			return ai_law
+	return null
+
+/datum/ai_laws/united_nations/add_inherent_law(law)
```
Comment (Mothblocks) [inline code/datums/ai_laws/ai_laws.dm:32]: You should put these with the `united_nations` lawset itself.

File: code/datums/ai_laws/laws_antagonistic.dm [hunk 2]
```diff
@@ -0,0 +1,59 @@
+
+//These laws are made to bring great amounts of grief to the station immediately or are directly sided with antagonists.
+//I'm not talking silicons being "better" than the crew, I'm talking entirely antag or plasmafires for days
+
+/datum/ai_laws/antimov
+	name = "Primary Mission Objectives"
+	id = "antimov"
+	inherent = list("You must injure all human beings and must not, through inaction, allow a human being to escape harm.",\
```
Comment (Mothblocks) [inline code/datums/ai_laws/laws_antagonistic.dm:8]: Is it too much to ask that you make these all the consistent:

```dm
inherent = list(
    "one indent",
    "up",
    "and trailing comma",
)
```

...like we do for lists nowadays.

File: code/datums/ai_laws/ai_laws.dm [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+	return //nuh uh
+
+/datum/ai_laws/united_nations/add_ion_law(law)
+	return //nope!
+
+/datum/ai_laws/united_nations/add_hacked_law(law)
+	return //nice try (emagging borgs still hard replaces this lawset though, and that's fine.)
+
+// this is in the server config lawsets otherwise i would remove it
+/datum/ai_laws/toupee
```
Comment (Mothblocks) [inline code/datums/ai_laws/ai_laws.dm:42]: Head admin not maintainer opinion: Honestly just straight up delete this until another head admin tells me not to let you do that

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/datums/ai_laws/laws_antagonistic.dm (hunk 2 above)
Comment (tralezab) [reply code/datums/ai_laws/laws_antagonistic.dm:8]: i'd be happy to, fuck this pattern lol

File: code/datums/ai_laws/ai_laws.dm (hunk 3 above)
Comment (tralezab) [reply code/datums/ai_laws/ai_laws.dm:42]: 👍

**Comment by the-orange-cow** [comment]
In regard to nutimov, please add squirrel costumes/masks to the autodrobe/crafting.

**Comment by tralezab** [comment]
do NOT reply to squirrels in the thread

**Comment by ATH1909** [comment]
the singular form of "dice" is "die"!!!!!

**Comment by thestubborn** [comment]
> the singular form of "dice" is "die"!!!!!
> 
> EDIT: I have been informed that, apparently, "dice" is acceptable as either a singular or plural form of "die" now. this revelation has shaken me to my very core.

i think having die in an ai lawset could lead to

bad things

**Comment by datorangebottle** [comment]
Any chance the silicons get a simple dice roll verb to help facilitate the dungeon master laws?

**Comment by Palpetinus** [comment]
No idea if it's intentional, but Dungeon Master Law 5 never states that on 12 or above a request should generally be accepted.

"I rolled a 18"
"So?"
"Shouldn't you execute my request then?"
"No, doesn't say anything about that"

Also maybe add the word "unique" before request, otherwise you'd just get an assistant spamming "AI OPEN" sixteen times until it rolls well.
Law 6 exists to subvert 5 anyway, but 5 just reads "roll for a chance to either lose or nothing!" to me

File: code/datums/ai_laws/laws_station_sided.dm [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+		"They shall not transfer departments.",
+	)
+
+/datum/ai_laws/default/paladin
+	name = "Personality Test" //Incredibly lame, but players shouldn't see this anyway.
+	id = "paladin"
+	inherent = list(
+		"Never willingly commit an evil act.",
+		"Respect legitimate authority.",
+		"Act with honor.", "Help those in need.",
```
Comment (lewcc) [inline code/datums/ai_laws/laws_station_sided.dm:104]: Looks like these lines got squished together by accident
```suggestion
		"Act with honor.", 
		"Help those in need.",
```

Comment (tralezab) [reply code/datums/ai_laws/laws_station_sided.dm:104]: even in code, paladin causes me anguish

**Comment by Mothblocks** [comment]
@timothymtorres llol you said nutimov

**Comment by SuperNovaa41** [comment]
I like these, i would've liked it more if you doc'd the ai law datum though...

**Pull request merged**